airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py

```diff
@@ -0,0 +1,239 @@
+            )
+
+    def job_get_id(self, response: Optional[requests.Response] = None) -> Optional[str]:
+        response_data = response.json() if response else {}
+        bulk_response = response_data.get("data", {}).get("bulkOperationRunQuery", {}).get("bulkOperation", {})
+        if bulk_response and bulk_response.get("status") == ShopifyBulkStatus.CREATED.value:
+            job_id = bulk_response.get("id")
+            self.logger.info(f"The BULK Job: `{job_id}` is {ShopifyBulkStatus.CREATED.value}")
+            return job_id
+        else:
+            return None
+
+    def job_retry_on_concurrency(self, request: requests.Request) -> Optional[requests.Response]:
+        if self.concurrent_attempt == self.concurrent_max_retry:
+            # indicate we're out of attempts to retry with job creation
+            self.concurrent_max_attempt_reached = True
+            self.logger.error(f"The BULK Job couldn't be created at this time, since another job is running.")
+            return None
+        elif self.concurrent_attempt < self.concurrent_max_retry:
+            # increment attempt
+            self.concurrent_attempt += 1
+            # try to execute previous request, it's handy because we can retry / each slice yielded
+            self.logger.warning(
+                f"The BULK concurrency limit has reached. Waiting {self.concurrent_interval_sec} sec before retry, atttempt: {self.concurrent_attempt}.",
+            )
+            sleep(self.concurrent_interval_sec)
+            # retry current `request`
+            return self.job_create(request)
+
+    def job_create(self, request: requests.Request) -> Optional[requests.Response]:
+        # create the server-side job
+        response = self.session.send(request)
+        # when the concurrent job takes place, we typically need to wait and retry, but no longer than 10 min.
+        if not self.has_running_concurrent_job(response):
+            # process created job response
+            return response if not self.job_check_for_errors(response) else None
+        else:
+            return self.job_retry_on_concurrency(request)
+
+    def log_job_status(self, bulk_job_id: str, status: str) -> None:
+        self.logger.info(f"The BULK Job: `{bulk_job_id}` is {status}.")
+
+    def job_check_status(self, url: str, bulk_job_id: str) -> Optional[str]:
+        request_args = {
+            "method": "POST",
+            "url": url,
+            "data": ShopifyBulkTemplates.status(bulk_job_id),
+            "headers": {"Content-Type": "application/graphql"},
+        }
+        status: str = None
+        while status != ShopifyBulkStatus.COMPLETED.value:
+            # re-use of `self._session(*, **)` to make BULK Job status checks
+            response = self.session.request(**request_args)
+            # errors check
+            errors = self.job_check_for_errors(response)
```

Comment:
`self.job_check_for_errors` doesn't return anything so errors will always be None. should this be calling `self.get_errors_from_response(response)`?